Issue a sane error message when crashing for three-dimensional svg write attempts

According to the `write_svg` documentation,

>    This function is currently only implemented for two-dimensional grids in two space dimensions

When I attempted to run this to visualize a 3D mesh, the program crashes, however, with a cryptic error message from the linker!

>Undefined symbols for architecture x86_64:
  "void dealii::GridOut::write_svg<3, 3>(dealii::Triangulation<3, 3> const&, std::__1::basic_ostream<char, std::__1::char_traits<char> >&) const", referenced from:
      first_grid() in step-1.cc.o
ld: symbol(s) not found for architecture x86_64

Since the linker error message wasn't exactly pinpointing on the spot, It took me a small bit of digging to figure out why `write_eps` worked while `write_svg` didn't. I mean, my debugging hat  didn't immeditaly think in this direction as I have visualised 3D svg output in a browser before coming to deal.II

I think that, following the exact idea on good programmign practices proposed in Prof. Bangerth's lecture, it may be wise to use the `Assert` macro with [ExcNotImplemented](https://dealii.org/developer/doxygen/deal.II/group__Exceptions.html#ga7b52b286796c23ef9ff178faf7a4b68f) or [ExcImpossibleInDim](https://dealii.org/developer/doxygen/deal.II/group__Exceptions.html#ga7b52b286796c23ef9ff178faf7a4b68f)  for returning a better error message.
______________________________

Additionally, I think it is a bit of a restriction that `ExcImpossibleInDim` can only take in one argument. In this case, we  know that only `dim=2` is a valid argument, and that `1` and `3` should produce errors. It would be better if this exception was made flexible to accept multiple arguments.

--- a/doc/news/changes/minor/20200114Bangerth
+++ b/doc/news/changes/minor/20200114Bangerth
@@ -1,0 +1,4 @@
+Fixed: The GridOut::write_svg() function was not instantiated for the
+case `dim=3, spacedim=3` by accident. This is now fixed.
+<br>
+(Wolfgang Bangerth, 2020/01/14)

--- a/source/grid/grid_out.inst.in
+++ b/source/grid/grid_out.inst.in
@@ -24,11 +24,6 @@ for (deal_II_dimension : DIMENSIONS)
       const Triangulation<deal_II_dimension> &, std::ostream &) const;
 #endif
 
-#if deal_II_dimension != 2
-    template void GridOut::write_svg(const Triangulation<deal_II_dimension> &,
-                                     std::ostream &) const;
-#endif
-
     template void GridOut::write_msh(const Triangulation<deal_II_dimension> &,
                                      std::ostream &) const;
 
@@ -72,13 +67,18 @@ for (deal_II_dimension : DIMENSIONS)
 
 for (deal_II_dimension : DIMENSIONS; deal_II_space_dimension : SPACE_DIMENSIONS)
   {
-#if deal_II_dimension < deal_II_space_dimension
-
-#  if deal_II_dimension != 2
+#if deal_II_dimension <= deal_II_space_dimension
+    // Instantiate write_svg exception for the <2,2> case for which there
+    // is an overload that takes care of that case.
+#  if !(deal_II_dimension == 2 && deal_II_space_dimension == 2)
     template void GridOut::write_svg(
       const Triangulation<deal_II_dimension, deal_II_space_dimension> &,
       std::ostream &) const;
 #  endif
+#endif
+
+
+#if deal_II_dimension < deal_II_space_dimension
 
     template void GridOut::write_msh(
       const Triangulation<deal_II_dimension, deal_II_space_dimension> &,

